lib/resources/port.rb

```diff
@@ -129,42 +135,102 @@ def info
   end
 end
 
-# extracts udp and tcp ports from macos
-class DarwinPorts < PortsInfo
-  def info
-    # collects UDP and TCP information
-    cmd = inspec.command('lsof -nP -iTCP -iUDP -sTCP:LISTEN')
-    return nil if cmd.exit_status.to_i != 0
+# extracts udp and tcp ports from the lsof command
+class LsofPorts < PortsInfo
+  attr_reader :lsof
+
+  def initialize(inspec, lsofpath = nil)
+    @lsof = lsofpath || 'lsof'
+    super(inspec)
+  end
 
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable MethodLength
```

Comment:
We should try not to disable the rubocup checks. I think we can cover most recommendations, once we extract the line parsing.